internal/executor/vaultunboxer/boxedvalue.go

```diff
@@ -0,0 +1,79 @@
+package vaultunboxer
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type BoxedValue struct {
+	vaultPath string
+	dataPath  []string
+}
+
+const (
+	prefix = "VAULT["
+	suffix = "]"
+)
+
+var (
+	ErrNotABoxedValue    = errors.New("doesn't look like a Vault-boxed value")
+	ErrInvalidBoxedValue = errors.New("Vault-boxed value has an invalid format")
+)
+
+func NewBoxedValue(rawBoxedValue string) (*BoxedValue, error) {
+	if !strings.HasPrefix(rawBoxedValue, prefix) || !strings.HasSuffix(rawBoxedValue, suffix) {
+		return nil, ErrNotABoxedValue
+	}
+
+	rawBoxedValue = strings.TrimPrefix(rawBoxedValue, prefix)
+	rawBoxedValue = strings.TrimSuffix(rawBoxedValue, suffix)
+
+	parts := strings.Split(rawBoxedValue, " ")
+	if len(parts) != 2 {
+		return nil, fmt.Errorf("%w: there should be 2 parameters (path and a selector), found %d",
+			ErrInvalidBoxedValue, len(parts))
+	}
+
+	dataPath := strings.Split(parts[1], ".")
+	if len(dataPath) == 0 {
```

Comment:
I think it's never true since `Split` will return at least a slice of one.

Reply:
See https://github.com/cirruslabs/cirrus-ci-agent/pull/253/commits/b37c902321ed7185cd20ec16b5325f5e31a07ec9.